### PR TITLE
chore: auto release to github & npmjs with browser detection

### LINKED ISSUE #81: Install from Github does not work
**Problem statement:**
Users should be able to install and use latest `master` version (or any git commit) through `npm i github:ethersphere/bee-js#branch_name`. However, this does not work right now because we are not building the library in the `prepack` script. Adding that is challenging, due to NPM bug https://github.com/npm/npm/issues/19564 where devDependencies (which we need for build) are not automatically installed.

**Task:**
- [ ] add `prepack` script to `package.json` which installs the necessary devDependencies and builds the project
- [ ] test it and demonstrate on a repo ideally running in CLI

### PR DESCRIPTION
**Pull request by vojtechsimetka**
This PR makes it easy to release the `bee-js` on github and npmjs. It also adds rewrite rule for webpack to automatically choose browser version when used in browser. You can see it in action my personal repo:

- [release PR](https://github.com/vojtechsimetka/bee-js/pull/19) with latest unreleased changes on `master` with two 'fix' commits
- [merged release PR](https://github.com/vojtechsimetka/bee-js/pull/17) that includes breaking change and the corresponding [publish on npmjs action](https://github.com/vojtechsimetka/bee-js/actions/runs/506012294)

also fixes #81 

## release-please action
Github action which runs with each new commit to `master` branch and creates or updates release PR. It automatically creates changelog based on the conventional commit messages and it also bumps the version. Currently it is configured to only take into account `feat` and `fix` keywords, but other can be configured if needed. The version in `package.json` and `package-lock.json` is automatically bumped with the following rules ([see more](https://github.com/googleapis/release-please#how-should-i-write-my-commits)):

- fix: which represents bug fixes, and correlates to a SemVer patch.
- feat: which represents a new feature, and correlates to a SemVer minor.
- feat!:, or fix!:, refactor!:, etc., which represent a breaking change (indicated by the !) and will result in a SemVer major.

The version number can be easily changed with a commit see [How do I change the version number?](https://github.com/googleapis/release-please#how-do-i-change-the-version-number)

After this PR gets merged, a new Github release with equivalent version is published. Since we are using personal access token, all the GH events are triggered (`push` to master, and the `release` events).

## publish action
When new Github release is published, a second Github action is run to compile browser and nodejs version of the code and publish to npmjs. 

### TODO: 
- [x] split the CI into two. One to publish on github, the other to publish on npmjs. This way it is less prone to an error and can be easily re-run
- [x] fix commit messages 
- [x] write documentation for the PR
- [x] finalize the name of the package and the npmjs organization

### CODE REVIEW

**Review by vandot — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "@ethersphere/bee-js",
+  "name": "@ethswarm/bee-js",
   "version": "0.1.0",
   "description": "Storage library for Bee",
```
Comment (agazso) [inline package.json:4]: I think a better description would be "Javascript client for Bee"

File: README.md [hunk 2]
```diff
@@ -32,35 +32,33 @@
 ### npm
 
 ```sh
-> npm install @ethersphere/bee-js
+> npm install @ethswarm/bee-js
```
Comment (agazso) [inline README.md:35]: Just an idea: if the package name doesn't follow strictly the github name, we could also name it as simply `@ethswarm/bee`  or `@ethswarm/bee-client`

**Review by agazso — COMMENTED** [review]
LGTM

File: package.json (hunk 1 above)
Comment (vojtechsimetka) [reply package.json:4]:
```suggestion
  "description": "Javascript client for Bee",
```

File: README.md [hunk 3]
```diff
@@ -32,35 +32,33 @@
...
+<script src="https://unpkg.com/@ethswarm/bee-js/dist/index.js"></script>
 ```
 
 ## Usage
 
 ```js
-import Bee from "@ethersphere/bee-js"; // Connect to a node const
+import Bee from "@ethswarm/bee-js"; // Connect to a node const
 
 bee = new Bee("http://localhost:8080");
```
Comment (vojtechsimetka) [inline README.md:63]:
```suggestion
bee = new Bee("http://localhost: 1633");
```

File: README.md (hunk 2 above)
Comment (vojtechsimetka) [reply README.md:35]: I'm adding that to the topics of tomorrow's JS team roundtable. We really should decide

File: package.json [hunk 4]
```diff
@@ -24,7 +24,7 @@
   "types": "dist/index.d.ts",
   "typings": "dist/index.d.ts",
   "scripts": {
-    "prepublishOnly": "rm -rf dist && npm run compile:types && npm run compile:browser && npm run compile:node",
+    "prepare": "rm -rf dist && npm run compile:types && npm run compile:browser && npm run compile:node",
```
Comment (agazso) [inline package.json:27]: This won't work on Windows. You may want to use the `rimraf` package as devDependency.

**Review by nugaon — APPROVED** [review]
LGTM!

**Pull request merged**